addons/account/static/src/js/bills_tree_upload.js

```diff
@@ -53,6 +53,12 @@ odoo.define('account.upload.bill.mixin', function (require) {
                 $formContainer = $(qweb.render('account.BillsHiddenUploadForm', {widget: this}));
                 $formContainer.appendTo(this.$('.o_content'));
             }
+            if (event.currentTarget.attributes.journal_type.value == 'sale'){
+                this.initialState.context['default_type'] = 'out_invoice'
+            }
+            if (event.currentTarget.attributes.journal_type.value == 'purchase'){
+                this.initialState.context['default_type'] = 'in_invoice'
+            }
```

Comment:
@william-andre You're right, passed `default_journal_id` in `context`. Also it will now generate document for appropriate journal when uploaded any document from invoice/bill tree view.

Reply:
It works as it should when coming from the dashboard or the list view, but I still have an issue. When I try to upload again from the "Generated Documents" list view, it doesn't keep the context and it goes in the default (Misc.) journal.

See the breadcrumbs and the journal in the screenshot below:
![Screenshot_2019-08-23  18 - Odoo](https://user-images.githubusercontent.com/44770049/63592778-bf8c3c00-c5b2-11e9-9942-b4f1b3c2beb2.png)

Reply:
@william-andre should be fixed with ba6ad5e515305cad52420999cbe151b7dcbd93d2

Reply:
Indeed